### PR TITLE
fix: no fallback to packages from csproj

### PR DESCRIPTION
**Pull request by aryehbeitz**
Two items:
1. the plugin is adding additional packages from the packages folder, even if not specified in source file.
2. additional packages should only be added if they exist in packages folder, but version is different
3. removing support for `.csproj` in favor or `packages.config`

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -178,15 +133,19 @@ module.exports = {
...
-              injectPath(dep);
-              flattendPackageList[dep.name] =
-                flattendPackageList[dep.name] || dep;
+              injectPath(dep, packagesFolder);
+              // only add a package from packages folder if version is different
+              if (flattendPackageList[dep.name] &&
+                flattendPackageList[dep.name].version !== dep.version) {
+                // prefer found from packages folder (dep) over existing
+                flattendPackageList[dep.name] = dep;
+              }
```
Comment (adrukh) [inline lib/index.js]: Let's add a `debug` message in this `if` clause to show which dep is involved, what version was extracted from the manifest file, and what version is overwriting it from the `packages` folder.

File: lib/index.js [hunk 2]
```diff
@@ -260,23 +219,33 @@ module.exports = {
...
+};
+
+function injectPath(dep, packagesFolder) {
+  dep.path =
+    dep.localPath ?
+    path.resolve(packagesFolder, dep.localPath)
+    : path.resolve(packagesFolder, dep.name + '.' + dep.version);
+  if (dep.localPath) {
+    delete dep.localPath;
+  }
```
Comment (adrukh) [inline lib/index.js]: Why is the deletion needed? This changes `dep` in a way that concurrent calls to this func will lead to different results. Seems 'surprising'?

Comment (aryehbeitz) [reply lib/index.js]: I'll take a look.
btw, I didn't write this function, just refactored it out from here https://github.com/snyk/snyk-nuget-plugin/pull/27/files#diff-6d186b954a58d5bb740f73d84fe39073L155

File: lib/index.js (hunk 1 above)
Comment (aryehbeitz) [reply lib/index.js]: updated.
```
snyk For package jQuery the version 1.1.1was extracted from manifest file.
snyk We are overwriting it with version 3.2.1 from the packages folder
```

**Pull request merged**